assets/js/base/components/text-input/index.js

```diff
@@ -14,6 +14,7 @@ import './style.scss';
 const TextInput = ( {
 	className,
 	id,
+	type,
```

Comment:
I wonder if it would be better to define the default value here?
```suggestion
	type = 'text',
```